### PR TITLE
add recipe for spot

### PR DESCRIPTION
**Pull request by adl**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by adl** [comment]
This package is currently available for Linux only on the [adl channel](https://anaconda.org/adl/spot). I used to build it by hand on by computer. But several users have reported issues such as glibc versioning issues (my build host has a more recent glibc than their computer) and lack of Darwin support.  I hope integrating spot in conda-forge will address both issues and make spot easier to find and install.

**Comment by adl** [comment]
@conda-forge-admin, please ping team
This is ready for review.

File: recipes/spot/build.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+./configure --prefix=$PREFIX --disable-devel --enable-optimizations
+make
+make install-strip
```
Comment (ocefpaf) [inline recipes/spot/build.sh:3]: Is there a `make check/test` we can run here?

Comment (adl) [reply recipes/spot/build.sh:3]: Yes, I can add `make check` there.

Comment (adl) [reply recipes/spot/build.sh:6]: Now the builds take 5 hours instead of 2.  (Our test suite is quite large.)

Comment (ocefpaf) [reply recipes/spot/build.sh:6]: Maybe we can run a subset of it? We do have 6 hours on Azure but I do not want to abuse it :smile: 

Anyway, let's leave that to the feedstock. Merging this!

**Pull request merged**